pkg/replica/replica.go

```diff
@@ -1156,3 +1177,72 @@ func (r *Replica) GetReplicaStat() (int64, int64) {
 
 	return lastModifyTime, headFileSize
 }
+
+func (r *Replica) GetFileInterval(includeBackingFileLayer bool) (sparse.GetFileInterval, error) {
+	if err := preload(&r.volume, includeBackingFileLayer); err != nil {
+		return nil, err
+	}
+	// baseDiskIndex is the smallest index in r.volume.files that is needed to export
+	// Note that index 0 in r.volume.files is nil. Index 1 is backing file if the volume has backing file.
+	// Otherwise, index 1 is regular disk file.
+	baseDiskIndex := 1
+
+	getFileInterval := func(ctx context.Context) (<-chan sparse.FileInterval, <-chan error, error) {
+		fileIntervalChannel := make(chan sparse.FileInterval)
+		errChannel := make(chan error)
```

Comment:
Can we use buffered channels?

Reply:
Why do we need to use buffered channels?
Is it because we have 4 worker gorotines that are taking the work from this channels?

Reply:
With a buffer, the goroutine can load the subsequential intervals when the caller is sending one interval. It makes the producing and consuming parallel.

Reply:
Thank you. Fixed